### ISSUE TITLE
Could not load file or assembly 'System.IdentityModel.Tokens.Jwt' in Azure Function V3

### ISSUE THREAD

**Issue by Chaos-Knight95** [pnp/pnpframework#352]
System.Private.CoreLib: Exception while executing function: Function1. PnP.Framework: Could not load file or assembly 'System.IdentityModel.Tokens.Jwt, Version=6.8.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

**Comment by jansenbe** [comment]
Are you using .NET 5? I've had issues with .NET 5 in V3 functions...can you try with .NET Core 3.1 and see if that works?

**Comment by Chaos-Knight95** [comment]
**..can you try with .NET Core 3.1 and see if that works?**
yes, I am using .net core 3.1 in the Azure function.

I have tried the PNP framework in the console application .net core  3.1 it is working fine there.
but when we are trying to work with the Azure function .net core 3.1 it is not working, we are facing the above-mentioned issue.

**Comment by jansenbe** [comment]
Did you cross check the System.IdentityModel.Tokens.Jwt assembly is correctly deployed to the Azure Function?

**Comment by Chaos-Knight95** [comment]
when I run the azure function in my local mechine it is not working

**Comment by jansenbe** [comment]
When you look in the bin folder, do you see System.IdentityModel.Tokens.Jwt.dll?

**Comment by Chaos-Knight95** [comment]
**When you look in the bin folder, do you see System.IdentityModel.Tokens.Jwt.dll?**
Yes, I am able to see that.

**Comment by jackpoz** [comment]
This issue seems to be the same as https://github.com/Azure/azure-functions-vs-build-sdk/issues/397 , with this resolution https://github.com/Azure/azure-functions-vs-build-sdk/issues/397#issuecomment-615445340

>For now, using `<_FunctionsSkipCleanOutput>true</_FunctionsSkipCleanOutput>` is the right approach to getting around this.

After that https://github.com/Azure/azure-functions-host/pull/6849 was merged introducting FunctionsPreservedDependencies as a csproj property to disable the cleanup.

I experienced this issue too in the past and solved it using `<_FunctionsSkipCleanOutput>true</_FunctionsSkipCleanOutput>` but I would suggest to try the "FunctionsPreservedDependencies" way.

tl;dr it's an issue of Azure Functions themselves, not pnp framework, but there are workarounds/solutions to get it to work that involve editing csproj of the Azure Function

**Comment by Chaos-Knight95** [comment]
it is working fine, Thank you for your help
